pkg/validate/sources.go

```diff
@@ -7,42 +7,54 @@ package validate
 import (
 	"context"
 	"fmt"
-	"net/url"
 
-	build "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	build "github.com/shipwright-io/build/pkg/apis/build/v1beta1"
 )
 
-// SourcesRef implements RuntimeRef interface to add validations for `build.spec.sources` slice.
-type SourcesRef struct {
+// SourcesRef implements RuntimeRef interface to add validations for `build.spec.source`.
+type SourceRef struct {
 	Build *build.Build // build instance for analysis
 }
 
-// ValidatePath executes the validation routine, inspecting the `build.spec.sources` path, which
-// contains a slice of BuildSource.
-func (s *SourcesRef) ValidatePath(_ context.Context) error {
-	for _, source := range s.Build.Spec.Sources {
-		if err := s.validateSourceEntry(source); err != nil {
-			return err
-		}
+// ValidatePath executes the validation routine, inspecting the `build.spec.source` path
+func (s *SourceRef) ValidatePath(_ context.Context) error {
+
+	if err := s.validateSourceEntry(s.Build.Spec.Source); err != nil {
+		return err
 	}
+
 	return nil
 }
 
 // validateSourceEntry inspect informed entry, probes all required attributes.
-func (s *SourcesRef) validateSourceEntry(source build.BuildSource) error {
-	if source.Name == "" {
-		return fmt.Errorf("name must be informed")
-	}
-	if source.URL == "" {
-		return fmt.Errorf("URL must be informed")
+func (s *SourceRef) validateSourceEntry(source build.Source) error {
+
+	// dont bail out if the Source object is empty, we preserve the old behaviour as in v1alpha1
+	if source.Type == "" && source.GitSource == nil &&
+		source.OCIArtifact == nil && source.LocalSource == nil {
+		return nil
 	}
-	if _, err := url.ParseRequestURI(source.URL); err != nil {
-		return err
+
+	switch source.Type {
+	case "Git":
+		if source.GitSource == nil {
+			return fmt.Errorf("type does not match the source")
+		}
```

Comment:
Do we want that additional check as well ?

```suggestion
		if source.GitSource == nil || source.OCIArtifact != nil || source.LocalSource != nil {
			return fmt.Errorf("type does not match the source")
		}
```